src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -410,7 +411,7 @@ static bool simplify_gather(std::shared_ptr<Node> node)
             // entire input tensor
             std::vector<int64_t> ref_indices(data.get_shape()[axis], 0);
             std::iota(ref_indices.begin(), ref_indices.end(), 0);
-            if (ref_indices == constant_indices->get_vector<int64_t>())
+            if (ref_indices == constant_indices->cast_vector<int64_t>())
```

Comment:
Curious, is there an issue with using get_vector?

Reply:
if indices is passed as i32, get_vector will conflict as we declare ref_indices as i64, so we explicitly cast